### ISSUE TITLE
Refactor dist build to improve maintainability

### ISSUE THREAD

**Issue by johnbotris** [tutao/tutanota#3221]
We have a single entry point into building the web client and the desktop apps, namely `dist.js`. Currently the prod build on Jenkins has 5 separate invocations to `dist.js`, each of which requires a very specific set of parameters, which in turn expect a specific workspace state, or else things will break (often only after having waited 20+ minutes). Any change to `dist.js` could potentially cause a breakage which could not possibly be noticed until actually trying to run the build.

Furthermore, `android.js` runs `dist.js` directly using `execFileSync`, forwarding on it's own parameters in a way that is fragile and currently broken anyway.

 In order to keep our build process more maintainable, and avoid spending time chasing breakages, we could take some of the following actions:
- [x] Split `dist.js` up into two distinct build scripts, `web.js` and `desktop.js`
- [x] Expose an API from `web.js` that can be called directly from both `desktop.js` or `android.js`
- [x] Move non-build steps out of the build script and into the Jenkinsfile (or some post-build script):
   * Desktop signing
   * Git tagging
   * `.deb` packaging
   * Downloading spellchecker dictionaries
   * Installing/publishing

Having a more well-defined build sequence should allow issue #2245 to be completed more easily.

**Comment by charlag** [comment]
We want to make a plan on refactoring build of different part of the app(s) and have a good concept so we will postpone and refactor this issue.

**Comment by rosso-ptg** [comment]
PR 3894 (https://github.com/tutao/tutanota/pull/3894) adds some improvements:
* Expose API for web app builds
* Split dist.js in separate scripts for desktop and webapp
* Code improvements/modularization

Some oddities that remain:
* We create index.html from the webapp build and then later overwrite it from the desktop build. Ideally, we would parametrize the webapp build, so that it would create the correct html file right away. In order to do so, some knowledge needs to be moved from a to b and the interface of the buildWebapp function needs be reworked.
* fetchDictionaries.js and publish.js contain a series of steps that would better be triggered from the Jenkinsfile directly rather than using javascript to invoke the command line.
* publish.js builds the .deb files for desktop and webapp and releases them - this should be split so that the release processes of the two can be independent of each other.

Once the above has been resolved, it would be very easy to split the Jenkinsfile and improve the release pipeline quite a bit.
